spec/requests/civil_servants/services_controller_spec.rb

```diff
@@ -0,0 +1,150 @@
+      end
+      let(:expected_service_strings) do
+        expected_services.flat_map do |service|
+          service_spec = service.service_specification
+          [
+            I18n.t('civil_servants.services.service_short_info_row.service_with', org_name: service.organization.name),
+            service_spec.name,
+            service_spec.location,
+            I18n.t('civil_servants.services.service_short_info_row.service_duration',
+                   beginning_date: I18n.l(service.beginning), ending_date: I18n.l(service.ending)),
+            I18n.t(service.confirmation_date.present?,
+                   scope: %i[civil_servants services service_short_info_row service_confirmed])
+          ]
+        end
+      end
+      let(:unexpected_service_strings) do
+        unexpected_services.flat_map do |service|
+          service_spec = service.service_specification
+          [
+            I18n.t('civil_servants.services.service_short_info_row.service_with', org_name: service.organization.name),
+            service_spec.name,
+            service_spec.location,
+            I18n.t('civil_servants.services.service_short_info_row.service_duration',
+                   beginning_date: I18n.l(service.beginning), ending_date: I18n.l(service.ending))
+          ]
+        end
+      end
+
+      before { sign_in civil_servant.user }
+
+      it 'returns http success' do
+        perform_request
+        expect(response).to have_http_status(:success)
+      end
+
+      context 'when the show_all filter is off' do
+        it 'renders the expected strings' do
+          perform_request
+          expect(response.body).to include(*expected_strings.map(&:to_s))
+          expect(response.body).not_to include(*unexpected_service_strings.map(&:to_s))
+        end
+      end
+
+      context 'when the show_all filter is on' do
+        let(:show_all) { true }
+        let(:expected_services) { [past_service, active_service, future_service] }
+        let(:unexpected_services) { [] }
+
+        it 'renders the expected strings' do
+          perform_request
+          expect(response.body).to include(*expected_strings.map(&:to_s))
+          expect(response.body).not_to include(*unexpected_service_strings.map(&:to_s))
+        end
+      end
+    end
+
+    context 'when an administrator is signed in' do
+      let(:organization_admin) { create :organization_member }
+
+      before do
+        sign_in organization_admin.user
+        perform_request
+      end
+
+      it_behaves_like 'unauthorized request'
+    end
+
+    context 'when no user is signed in' do
+      before { perform_request }
+
+      it_behaves_like 'unauthenticated request'
+    end
+  end
+
+  describe '#show' do
+    let(:perform_request) { get civil_servants_service_path(service) }
+
+    let(:service) { create :service }
+
+    context 'when a civil servant is signed in' do
+      let(:civil_servant) { service.civil_servant }
+
+      before { sign_in civil_servant.user }
```

Comment:
You don't use the let but in the before

```suggestion
      before { sign_in service.civil_servant.user }
```